crates/vm/vm.rs

```diff
@@ -75,34 +75,172 @@ impl EvmState {
+                cumulative_gas_used += result.gas_used;
+                let receipt = Receipt::new(
+                    transaction.tx_type(),
+                    matches!(result.result, TxResult::Success),
+                    cumulative_gas_used,
+                    // TODO: map our logs to the logs expected by ethereum_rust
+                    vec![],
+                );
+                receipts.push(receipt);
 
-    if let Some(withdrawals) = &block.body.withdrawals {
-        process_withdrawals(state, withdrawals)?;
-    }
+                for (address, account) in result.new_state {
+                    let mut added_storage = HashMap::new();
+
+                    for (key, value) in account.storage {
+                        added_storage.insert(key, value.current_value);
+                    }
+
+                    let code = if account.info.bytecode.is_empty() {
+                        None
+                    } else {
+                        Some(account.info.bytecode.clone())
+                    };
+
+                    let account_update = AccountUpdate {
+                        address,
+                        removed: false,
+                        info: Some(AccountInfo {
+                            code_hash: code_hash(&account.info.bytecode),
+                            balance: account.info.balance,
+                            nonce: account.info.nonce,
+                        }),
+                        code,
+                        added_storage,
+                    };
+
+                    account_updates.push(account_update);
+                }
+            }
+
+            if let Some(withdrawals) = &block.body.withdrawals {
+                process_withdrawals(state, withdrawals)?;
+            }
+
+            Ok((receipts, account_updates))
+        }
+
+        pub fn execute_tx_levm(
+            tx: &Transaction,
+            block_header: &BlockHeader,
+            db: Arc<dyn LevmDatabase>,
+        ) -> Result<TransactionReport, VMError> {
+            dbg!(&tx.tx_type());
+
+            let gas_price: U256 = match tx.tx_type() {
+                TxType::Legacy => tx.gas_price().into(),
+                TxType::EIP2930 => tx.gas_price().into(),
+                TxType::EIP1559 => {
+                    let priority_fee_per_gas = min(
+                        tx.max_priority_fee().unwrap(),
+                        tx.max_fee_per_gas().unwrap() - block_header.base_fee_per_gas.unwrap(),
+                    );
+                    (priority_fee_per_gas + block_header.base_fee_per_gas.unwrap()).into()
+                }
+                TxType::EIP4844 => {
+                    let priority_fee_per_gas = min(
+                        tx.max_priority_fee().unwrap(),
+                        tx.max_fee_per_gas().unwrap() - block_header.base_fee_per_gas.unwrap(),
+                    );
+                    (priority_fee_per_gas + block_header.base_fee_per_gas.unwrap()).into()
```

Comment:
Keep in mind: This operations use u64s, what happens if there is an overflow? Can it happen? What should we do in such case? Depending on the behavior we want, we can use a dedicated operation for this like saturating_add, wrapping_add, etc..

Reply:
Agreed, tracked this in https://github.com/lambdaclass/lambda_ethereum_rust/issues/1075, it'll be done after removing the `unwrap`s and `expect`s so we have a solid error handling basis.